### PR TITLE
fix: [CDS-48188]: generic tag fix for custom deployment 

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- `isGenericArtifactory` check was missing for customDeployment which is being used to render artifactoryDirectory, tags field, and to fetch repositories list based on repository format.
- Fixes validation schema for artifacts
- Update default total visible instances count

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by Sagar-Rajput7 — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx [hunk 1]
```diff
@@ -275,9 +278,29 @@ const Content = (props: ArtifactoryRenderContent): JSX.Element => {
...
   }, [service, artifactPath, selectedDeploymentType, initialValues, artifact])
 
+  const isCustomDeploymentGenericSelected = useMemo(() => {
+    /* istanbul ignore else */
+    if (service) {
+      const parsedService = service?.data?.yaml && parse(service?.data?.yaml)
+      // to be refactored for some fields once generic dependency is resolved via V2
+      const artifactsInfo = get(parsedService, `service.serviceDefinition.spec.artifacts`) as ArtifactListConfig
+      artifactsInfo?.primary?.sources?.map(artifactInfo => {
+        if (artifactInfo?.identifier === (artifact as ArtifactSource)?.identifier) {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:288]: This component is rendered for each artifact source, if I am not missing anything. Then why this change is needed ? Can't we directly use `artifact.spec.repositoryFormat` ?

Comment (PratyushGar) [reply src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/ArtifactoryArtifactSource/ArtifactoryArtifactSource.tsx:288]: Artifact has only runtime fields, so repositoryFormat is being calculated from the service data

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]
